### PR TITLE
Factory Method implementation (+ app and database initialization changes)

### PR DESCRIPTION
**Pull request by KhrystynaVysotska**
**Changes:**

- introduced Marshmallow to the project; initialized and created schemas for book and library item
- changed the app and database initialization:
    - moved instances of Marshmallow and SQLAlchemy to the shared resources.py file, 
    - initialized them with the app in main.py
    - setup auto-initialization of db in main.py
    - deleted db.py file and removed it from the "exclude" block of setup.cfg
    - modified models to use SQLAlchemy instance from shared resources.py file
- implemented factory method for library items creation and other operations
- modified book controller to work with real data (removed mocks)

**Screens on how book_controller works:**

_**GET ALL**_
image

_**GET BY ID (exists)**_
image

_**GET BY ID (not exists)**_
image

_**POST**_
image

**_PUT (exists)_**
image

**_PUT (not exists)_**
image

**_DELETE (exists)_**
image

**_DELETE (not exists)_**
image

### CODE REVIEW

File: models/book.py [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+    )
+    title = db.Column(db.String(50), unique=True)
+    author = db.Column(db.String(60))
+    genre = db.Column(db.Enum(BookGenre))
 
     def __repr__(self):
         return str(self.__dict__)
+
+    @staticmethod
+    def from_dict(body: dict):
```
Comment (Ari100telll) [inline models/book.py:19]: Better to use .get("key", None) instead of ["key"], because ["key"] can raise KeyError

File: models/library_item.py [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-    pledgePrice = Column(Numeric)
-    quantity = Column(Integer)
+    id = db.Column(db.Integer, primary_key=True)
+    pledge_price = db.Column(db.Numeric)
+    quantity = db.Column(db.Integer)
 
     def __repr__(self):
         return str(self.__dict__)
 
+    @staticmethod
```
Comment (Ari100telll) [inline models/library_item.py:16]: Better to use .get("key", None) instead of ["key"], because ["key"] can raise KeyError

**Review by Ravenen — DISMISSED** [review]

File: models/book.py [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
+        title = body.get("title", None)
+        author = body.get("author", None)
+        genre = (body.get("genre", None) or "").upper()
```
Comment (Ari100telll) [inline models/book.py]: get("genre", "")

File: services/factory/book_factory.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    def update_library_item(self, item_id: int, body: dict) -> LibraryItem:
+        library_item_update_body = {
+            "pledge_price": body.get("pledge_price", None),
+            "quantity": body.get("quantity", None),
+        }
+        book_update_body = {
+            "author": body.get("author", None),
+            "title": body.get("title", None),
+            "genre": (body.get("genre", None) or "").upper(),
```
Comment (Ari100telll) [inline services/factory/book_factory.py]: body.get("genre", "")

**Review by Ari100telll — DISMISSED** [review]

**Review by Ravenen — APPROVED** [review]

**Review by Ari100telll — APPROVED** [review]

**Pull request merged**